internal/clients/kube/apply_client.go

```diff
@@ -132,19 +139,37 @@ func (a ApplyClient) deleteObject(ctx context.Context, mapping *meta.RESTMapping
 
 func (a ApplyClient) applyObject(ctx context.Context, mapping *meta.RESTMapping, applyObject applyObject) error {
 	if isClusterScopedResource(mapping.Resource.Resource) {
+		// Don't risk overwriting a namespace if it already exists
+		if mapping.Resource.Resource == "namespaces" {
+			_, err := a.dynamicClient.Resource(mapping.Resource).Get(ctx, applyObject.name, v1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					_, err = a.dynamicClient.Resource(mapping.Resource).Apply(ctx, applyObject.name, &unstructured.Unstructured{Object: applyObject.unstructuredObj}, v1.ApplyOptions{FieldManager: "application/apply-patch"})
+					return err
+				}
+
+				return err
+			}
+
+			// Object already exists, just patch the namespace metadata.
+			metadata := applyObject.unstructuredObj["metadata"]
```

Comment:
Should we add a logger.info for this case to be able to identify this in "normal" logs?